### PR TITLE
Increasing Prometheus retention time from default 15d to 1 month after which old data will be removed

### PR DESCRIPTION
**Pull request by ibuziuk**

### CODE REVIEW

File: openshift/che-monitoring.yaml [hunk 1]
```diff
@@ -76,6 +76,7 @@ objects:
           - -upstream=${PROMETHEUS_SVC_URL}
           - -github-org=app-sre
           - -github-team=app-sre-observability
+          - --storage.tsdb.retention.time=30d
```
Comment (aditya-konarde) [inline openshift/che-monitoring.yaml]: This is the wrong container to add this to. 

We need to add it to the prometheus container, the one you've added it to is the reverse proxy that does authz/n sitting in front of prometheus

**Review by aditya-konarde — CHANGES_REQUESTED** [review]

File: openshift/che-monitoring.yaml (hunk 1 above)
Comment (ibuziuk) [reply openshift/che-monitoring.yaml]: good catch

**Comment by ibuziuk** [comment]
@aditya-konarde PR has been updated

**Comment by skabashnyuk** [comment]
Just curious is there any reason no to set 1year (ok let's say 3 months)? As for me, the efforts of storing this data is not comparable to analytical results that can be done in a long period trend.

**Comment by ibuziuk** [comment]
@skabashnyuk I talked about it with @aditya-konarde but was told that setting 1 year retention period would create additional load on storage and RAM. So, if we want to increase the retention period we need to update RAM and Storage config accordingly

**Comment by skabashnyuk** [comment]
ok

**Review by aditya-konarde — APPROVED** [review]
lgtm

**Comment by aditya-konarde** [comment]
An additional point to answer @skabashnyuk 's question is that long term retention shouldn't be handled at the Prometheus layer. Thanos fits exactly that piece of the puzzle, and lets you do a much better optimization on distributing your queries so that they don't OOM your prometheus itself. 

We should prioritize a move to 4.x for Che, which will allow us to benefit from a properly deployed Prometheus stack that we'll integrate with a thanos instance. 

Until then, I think we should not try to keep the toil on this stop-gap Prometheus instance as low as possible.

**Comment by ibuziuk** [comment]
> We should prioritize a move to 4.x for Che, which will allow us to benefit from a properly deployed Prometheus stack that we'll integrate with a thanos instance.

cc: @pbergene am I correct that the plan of dsaas / dsas-stg 4.x update is some time this year?

**Comment by pbergene** [comment]
@ibuziuk Last we visited this we were trying to find what time this work could start on the devtools side.  I would absolutely expect this year.

**Pull request merged**